src/lua/minifio.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2023, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#include "lua/minifio.h"
+
+#include <unistd.h>
+#include <lua.h>
+#include <lauxlib.h>
+
+#include "diag.h"
+#include "lua/utils.h"
+#include "lua/error.h"
+
+/**
+ * Push nil and an error with strerror() based message.
+ */
+static int
+luaT_minifio_pushsyserror(struct lua_State *L)
+{
+	/*
+	 * It is used in functions exposed into the fio module,
+	 * so use "fio" diagnostics instead of "minifio".
+	 */
+	diag_set(SystemError, "fio");
+	return luaT_push_nil_and_error(L);
+}
+
+/**
+ * minifio.cwd() -- get current working directory.
+ */
+static int
+lbox_minifio_cwd(struct lua_State *L)
+{
+	char *buf = (char *)lua_newuserdata(L, PATH_MAX);
+	if (!buf) {
+		errno = ENOMEM;
+		return luaT_minifio_pushsyserror(L);
+	}
+	if (getcwd(buf, PATH_MAX) == NULL)
```

Comment:
Your points look correct on a first glance, but I would answer here as [above](https://github.com/tarantool/tarantool/pull/8250#discussion_r1102729134): let's work on it somehow separately. Here I'm moving the code. I understand the 'while we're here' reason, but actually I don't know why it is written in the way it is written. I should look down to the history and test different variants. It is not like 'fix a typo in a comment', at least for me.

Reply:
Alright, that works for me too.

What about "why it is written in this way", I doubt it's necessary to allocate an auxiliary buffer instead of using the static one, but `lua_newuserdata` **never** yields `NULL`, so the latter is simply a bug.

All in all, resolving.

Reply:
I [added](https://www.notion.so/tarantool/fio-cwd-implementation-oddity-a9314c7951bd495281da6de4beffd4f0) it to the wishlist.